### PR TITLE
[rewrite] remove dependency on stream in stream helpers

### PR DESCRIPTION
**Pull request by schtauffen**
combine is available on any stream's dot-constructor

### CODE REVIEW

**Comment by isiahmeadows** [comment]
Logically, the dependency will still exist. It's just a matter of where.

**Comment by schtauffen** [comment]
It is a softer dependency. 

> I require **this** implementation

vs 

> I require any implementation of form (...)

I haven't checked but had assumed this change would reduce boilerplate output by webpack.
I also tend to prefer requiring interfaces over specific implementations.

File: stream/scanMerge.js [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-
 module.exports = function(tuples, seed) {
 	var streams = tuples.map(function(tuple) {
 		var stream = tuple[0]
 		if (stream._state.state === 0) stream(undefined)
 		return stream
 	})
 
-	var newStream = combine(function() {
+	var newStream = streams[0].constructor.combine(function() {
```
Comment (lhorie) [inline stream/scanMerge.js]: streams[0] could be undefined

Comment (schtauffen) [inline stream/scanMerge.js]: Then it will error. I would say that is working as intended :)

**Pull request closed without merging**